### ISSUE TITLE
[BUDI-6868] Generated CRUD screens will update but not create (SAVE ROW does not execute)

### ISSUE THREAD

**Issue by simon-lewis** [Budibase/budibase#10322]
**Checklist**
- [ X ] I have searched Budibase discussions and Github issues to check if my issue already exists
- Have searched for any similar issues with MS SQL or CRUD screens generally

**Hosting**
- Self
  - Method: docker compose
  - Budibase Version: 2.5.2
  - App Version: 2.5.2

**Describe the bug**
Have several apps in Budibase and generally they work well. The new Side Panel-based CRUD screen generation is good but I am experiencing a problem with the Create New Row Side Panel as generated. Working with an MS SQL Server database, I can create rows in the Data Browser within the Builder (I can add rows). When I generate a table-based CRUD screen with side panels, the New Row side panel does not action the save and does not execute any subsequent actions, including closing/hiding the side panel. I assume there is some kind of error when I click the button (actually when the SAVE ROW action attempts to execute) but I cannot surface that error or issue. I have ejected the components to analyse the design and actions but no joy.

I have checked the Data definition / DB connection and it _can_ create rows - I have managed to do this from the Data tab in the Builder. 

The behavior suggests a privilege error or missing data in the form that prevents the record being saved but I cannot detect this. Have checked the docker-compose logs to see if I can spot an error but no joy.

**To Reproduce**
1. Generate a CRUD table screen using wizard in Builder as normal. 
2. Publish the app and use the new screen to view records and update a row - this works as normal
3. Use the new screen to ADD a row - this does not work and appears to stop at the SAVE ROW action
4. Activate (set) the Confirmation option in the Save Row action and retry - this prompts the user to confirm but clicking OK then produces no further action. The row is not saved and the modal form (or side bar) is not closed - the actions appear to stop processing.

**Expected behavior**
The SAVE ROW action pops up a confirmation (if selected) and saves the form data as a new row in the table OR it prompts with any errors.

**Screenshots**
Attached (not particularly useful)
![BBIssue-PostSave](https://user-images.githubusercontent.com/92350835/232376590-74172256-88d1-44e8-b771-da41b5b242f6.png)
![BBIssue-PreSave](https://user-images.githubusercontent.com/92350835/232376591-139d3386-ac8a-475b-b208-e352b141bec0.png)

  
**App Export**
Not possible to share as includes confidential data

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Edge
 - Version: 112.0.1722.39 (Official build) (64-bit)

**Additional context**
Accessing on-prem SQL Server - connectivity is good; update actions work fine. There are no constraints which should prevent row creation (and I can in fact create rows using the Data section of the builder portal).

<sub>[BUDI-6868](https://linear.app/budibase/issue/BUDI-6868/generated-crud-screens-will-update-but-not-create-save-row-does-not)</sub>

**Comment by mike12345567** [comment]
Hi @simon-lewis - can you look to see what the response is from the network request to save the row? This may be throwing an error of some kind.

**Comment by simon-lewis** [comment]
Hi @mike12345567 - yep, checked into this. Ran SQL Profiler and trapped SELECT and UPDATE statements coming from Budibase. Can't see any INSERT statements (or any other activity when the SAVE ROW action is, er, actioned). If I use the Data designer in the budibase Builder and click on Create Row, I can add a row and obviously see this hitting the SQL server as an INSERT.
(Seeing "**insert into [tblProcess] ([blnValid], [blnWatch], [lngApplicationID], [lngProcessTypeID], [strAuthor], [strProcessName]) values (@p0, @p1, @p2, @p3, @p4, @p5)**" coming from Builder 'create row'. Nothing from SAVE ROW action in running app.).
In short, looks like the INSERT statement is not being sent by Budibase from within an app SAVE ROW action but I can't see why (no error message returned and subsequent ACTION steps do not run).
Probably unrelated but we are also seeing intermittent errors  - "Unable to fully release the lock on resource 'lock:default_persist_writethrough_usage_quota'". "

**Comment by jmlaurenti** [comment]
Hi @simon-lewis and @mike12345567 ,
I had the same issue yesterday (with mySQL tables) with 2.5.2, fallback to 2.4.31 fixed the problem.
Cannot share any log, just wanted to highlight the issue seems to be in a change  included in latest release.
Thanks, JML

**Comment by mike12345567** [comment]
> Hi @mike12345567 - yep, checked into this. Ran SQL Profiler and trapped SELECT and UPDATE statements coming from Budibase. Can't see any INSERT statements (or any other activity when the SAVE ROW action is, er, actioned). If I use the Data designer in the budibase Builder and click on Create Row, I can add a row and obviously see this hitting the SQL server as an INSERT. (Seeing "**insert into [tblProcess] ([blnValid], [blnWatch], [lngApplicationID], [lngProcessTypeID], [strAuthor], [strProcessName]) values (@p0, @p1, @p2, @p3, @P4, @p5)**" coming from Builder 'create row'. Nothing from SAVE ROW action in running app.). In short, looks like the INSERT statement is not being sent by Budibase from within an app SAVE ROW action but I can't see why (no error message returned and subsequent ACTION steps do not run). Probably unrelated but we are also seeing intermittent errors - "Unable to fully release the lock on resource 'lock:default_persist_writethrough_usage_quota'". "
> 
> ![bberror-19Apr23](https://user-images.githubusercontent.com/92350835/232901470-1994f9a6-58c2-4d7a-af2b-129ab6383f84.png)

Hi @simon-lewis - would you be able to confirm if a request is being sent off from the app? If you open the developer tools in the browser you should be able to check out the network tab and see if any request is being sent when you run the save row action in an app.

**Comment by simon-lewis** [comment]
@aptkingston  Hi Andrew - here you go. This is a fairly common action - we use it to record access to the app. Its a single SAVE ROW action to a SQL Server dataset. The same issue applies to generated actions - the SAVE ROW step for a generated NEW ROW SIDE BAR also fails and the subsequent steps do not execute. See comment from @jmlaurenti Jean-Michel above. Looks like an issue introduced after 2.4.31. This squares with what I have seen - we stopped getting these activity records being created on or around 5th April which probably coincided with an update applied to BB.
![bberror-20Apr23](https://user-images.githubusercontent.com/92350835/233180251-d9a2ddce-380b-4c65-8683-f8b0453b2c03.png)

**Comment by mike12345567** [comment]
Hi @simon-lewis - if possible it would be great as well to see the network logs from the developer console in the browser when you click the button action, to see if there is a request being sent at all.

**Comment by simon-lewis** [comment]
@mike12345567 Hi Mike - good call as this shows a SQL error being returned within a status 500 response. In this case, it is saying a specific field cannot be blank - fine but this is new behaviour as it worked previously. The field in question must indeed  be NOT NULL on the SQL server BUT has a default value so this should not be an issue (field is a date field, "dtmEvent" and is not explicitly populated in the SAVE ROW action). I've attached 2 examples - the other shows a similar error on relationship fields: again, this worked fine under previous versions as the relationship dropdowns on the input (create row) form satisfied the NOT NULL constraint on the underlying foreign key field. Hope that makes sense - again, the behaviour has changed.

[budibase-GenericSaveRowAction-BrowserNetworkLog-20Apr23.har.txt](https://github.com/Budibase/budibase/files/11277189/budibase-GenericSaveRowAction-BrowserNetworkLog-20Apr23.har.txt)
[budibase-InitialSaveRowActionOnLaunchs-BrowserNetworkLog-20Apr23.har.txt](https://github.com/Budibase/budibase/files/11277190/budibase-InitialSaveRowActionOnLaunchs-BrowserNetworkLog-20Apr23.har.txt)

**Comment by mike12345567** [comment]
Hi @simon-lewis - ah looks like we've got to the bottom of this. There was a bug fix as Budibase wasn't validating for Not null checks in the UI, but we did find there was an issue with default values on fields - we've fixed that and it will be released soon.

https://github.com/Budibase/budibase/pull/10312

**Comment by aptkingston** [comment]
I'm a bit confused because if you were getting HTTP 500s with validation errors, you would be getting error popups in your app, like so:
![image](https://user-images.githubusercontent.com/9075550/233307232-6b582de6-839c-4c9f-8244-2698b9cc0240.png)

Are you sure there is no other error being thrown in your console? Unless something else is crashing the JS on your page (which would show an error in the console) you would be getting error popups here showing the validation failures. The output of your browser console would be very helpful here, e.g.:
![image](https://user-images.githubusercontent.com/9075550/233307842-513d2c01-5d24-4be2-bb40-1b58c215b388.png)

There would need to be some other unhandled JS error, or else you would be seeing the validation popups.

**Comment by simon-lewis** [comment]
@aptkingston - Thanks Andrew - yeah, was sure. Nothing popping up. Looks like @mike12345567 Mike has found the cause and got it on the fix list. Thanks so much for the time / assist on this one. Will keep an eye out for the update under https://github.com/Budibase/budibase/pull/10312.
